### PR TITLE
Feature/web documentation

### PR DESCRIPTION
**Pull request by Louismaestre**
#332 

Create the web documentation : dart doc .
Display the documentation : open doc/api/index.html 

Landing page for the documentation : 

Capture d’écran 2024-05-17 à 23 59 13

Screen of a page :

 - Admin Page : 

Capture d’écran 2024-05-17 à 23 59 25

Capture d’écran 2024-05-17 à 23 59 28

 - App route Page : 

Capture d’écran 2024-05-17 à 23 59 33

Capture d’écran 2024-05-17 à 23 59 37

 - Autofill dialog : 
 
Capture d’écran 2024-05-18 à 00 03 58

Capture d’écran 2024-05-18 à 00 05 23

### CODE REVIEW

File: Configurateur/front/lib/main.dart [hunk 1]
```diff
@@ -6,6 +6,7 @@ import './app_routes.dart';
 import 'styles/themes.dart';
 import 'services/theme_service.dart';
 
+/// [Function] : Lunch the web application.
```
Comment (H-chauvet) [inline Configurateur/front/lib/main.dart:9]: Are you hungry ? (Launch, not lunch)

**Review by H-chauvet — CHANGES_REQUESTED** [review]

**Review by H-chauvet — APPROVED** [review]

File: Configurateur/front/lib/components/container.dart [hunk 2]
```diff
@@ -7,6 +7,18 @@ import 'package:front/services/storage_service.dart';
...
 import 'package:go_router/go_router.dart';
 
+/// ContainerListData
+///
+/// Define the data of container in back end
+/// [id] : Container's id
+/// [createdAt] : Creation of the container
+/// [organization] : Organization having created the container
+/// [organizationId] : Id of the organization
+/// [containerMapping] : ???
```
Comment (31Nathan) [inline Configurateur/front/lib/components/container.dart:17]: ???? Can not leave that undocumented

File: Configurateur/front/lib/components/custom_header.dart [hunk 3]
```diff
@@ -7,23 +7,31 @@ import 'package:front/styles/themes.dart';
...
+/// Header for the web pages
 class LandingAppBar extends StatefulWidget {
   const LandingAppBar({super.key, required BuildContext context});
 
   @override
   State<LandingAppBar> createState() => LandingAppBarState();
 }
 
+/// LandingAppBarState
+///
```
Comment (31Nathan) [inline Configurateur/front/lib/components/custom_header.dart:21]: Missing something between [ ]  ?

File: Configurateur/front/lib/components/dialog/rating_dialog_content/rating_dialog_content.dart [hunk 4]
```diff
@@ -74,6 +81,9 @@ void sendData(String rating, String message) async {
   }
 }
 
+/// RatingDialogContent
```
Comment (31Nathan) [inline Configurateur/front/lib/components/dialog/rating_dialog_content/rating_dialog_content.dart:84]: Missing a thing between [ ]  ?

File: Configurateur/front/lib/components/recap_panel/recap_panel.dart [hunk 5]
```diff
@@ -32,6 +46,7 @@ class LockerList {
 ///
 /// RecapPanel
 ///
+/// Summary of the lockers selected in a container
```
Comment (31Nathan) [inline Configurateur/front/lib/components/recap_panel/recap_panel.dart:49]: Missing the [ ] thing

File: Configurateur/front/lib/screens/admin/admin.dart [hunk 6]
```diff
@@ -11,20 +11,28 @@ import 'package:front/styles/themes.dart';
...
+/// Page only for the admin to manage the database easier
 class AdminPage extends StatefulWidget {
   const AdminPage({Key? key}) : super(key: key);
 
   @override
   _AdminPageState createState() => _AdminPageState();
 }
 
+/// AdminPageState
+///
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/admin/admin.dart:25]: Missing [ ] thing ?

File: Configurateur/front/lib/screens/company-profil/company-profil.dart [hunk 7]
```diff
@@ -54,6 +63,9 @@ class OrganizationList {
   }
 }
 
+/// CompanyProfilPage
+///
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/company-profil/company-profil.dart:67]: Missing the [ ] thing

File: Configurateur/front/lib/screens/company-profil/container-profil.dart [hunk 8]
```diff
@@ -20,10 +24,12 @@ class ContainerProfilPage extends StatefulWidget {
       _ContainerProfilPageState(container: container);
 }
 
+/// CompanyProfilPageState
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/company-profil/container-profil.dart:27]: Missing [ ] thing ?

File: Configurateur/front/lib/screens/company/company.dart [hunk 9]
```diff
@@ -15,15 +15,18 @@ import 'package:http/http.dart' as http;
...
+///
+/// Page of the Risu Company and the containers created with the configurator
 class CompanyPage extends StatefulWidget {
   const CompanyPage({Key? key}) : super(key: key);
 
   @override
   State<CompanyPage> createState() => CompanyPageState();
 }
 
+/// CompanyPageState
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/company/company.dart:28]: Missing [ ]

File: Configurateur/front/lib/screens/confidentiality/confidentiality.dart [hunk 10]
```diff
@@ -2,9 +2,13 @@ import 'package:flutter/material.dart';
 import 'package:front/components/custom_app_bar.dart';
 import 'package:front/components/footer.dart';
 
+/// ConfidentialityPage
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/confidentiality/confidentiality.dart:5]: Missing []

File: Configurateur/front/lib/screens/contact/contact.dart [hunk 11]
```diff
@@ -15,13 +15,17 @@ import 'package:front/styles/themes.dart';
 import 'package:http/http.dart' as http;
 import 'package:provider/provider.dart';
 
+/// ContactPage
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/contact/contact.dart:18]: Missing [ ]

File: Configurateur/front/lib/screens/contact/contact.dart [hunk 12]
```diff
@@ -61,13 +65,16 @@ void sendFormData(
   }
 }
 
+/// ContactPageState
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/contact/contact.dart:68]: Same

File: Configurateur/front/lib/screens/container-creation/confirmation_screen/confirmation_screen.dart [hunk 13]
```diff
@@ -6,18 +6,20 @@ import 'package:go_router/go_router.dart';
 
 import 'confirmation_screen_style.dart';
 
+/// ConfirmationScreen
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/confirmation_screen/confirmation_screen.dart:9]: Missing [ ]

File: Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart [hunk 14]
```diff
@@ -43,10 +50,8 @@ class ContainerCreation extends StatefulWidget {
   State<ContainerCreation> createState() => ContainerCreationState();
 }
 
+/// ContainerCreationState
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart:53]: []

File: Configurateur/front/lib/screens/container-creation/design_screen/design_screen.dart [hunk 15]
```diff
@@ -33,6 +34,9 @@ const List<String> faceList = <String>[
   'Bas'
 ];
 
+/// Design for a container
+/// [face] : Contient les face du conteneur
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/design_screen/design_screen.dart:38]: In english or in french but choose

File: Configurateur/front/lib/screens/container-creation/maps_screen/maps_screen.dart [hunk 16]
```diff
@@ -33,6 +41,8 @@ class MapsScreen extends StatefulWidget {
   State<MapsScreen> createState() => MapsState();
 }
 
+/// MapState
+///
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/maps_screen/maps_screen.dart:45]: []

File: Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart [hunk 17]
```diff
@@ -35,11 +43,14 @@ class PaymentScreen extends StatefulWidget {
   State<PaymentScreen> createState() => _PaymentScreenState();
 }
 
+/// _PaymentScreenState
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/payment_screen/payment_screen.dart:46]: []

File: Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart [hunk 18]
```diff
@@ -31,14 +39,12 @@ class RecapScreen extends StatefulWidget {
   State<RecapScreen> createState() => RecapScreenState();
 }
 
-///
-/// Login screen
-///
-/// page de connexion pour le configurateur
+/// RecapScreenState
```
Comment (31Nathan) [inline Configurateur/front/lib/screens/container-creation/recap_screen/recap_screen.dart:42]: []

**Review by 31Nathan — CHANGES_REQUESTED** [review]
Check everything please, the same thing is on almost all your files

File: Configurateur/front/lib/components/container.dart (hunk 2 above)
Comment (CedricCORGE) [reply Configurateur/front/lib/components/container.dart:17]: it's a string that contains numbers representing where lockers is positioned in the container.

File: Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart [hunk 19]
```diff
@@ -501,6 +518,9 @@ class ContainerCreationState extends State<ContainerCreation> {
     }
   }
 
+  /// [Function] : Delete a locker
+  /// [coord] : the locker's position in the container
+  /// [unitTesting] : ???
```
Comment (CedricCORGE) [inline Configurateur/front/lib/screens/container-creation/container_creation/container_creation.dart:523]: boolean that says if we came here from unit test or not

**Review by 31Nathan — APPROVED** [review]

**Pull request merged**